Prevent Tab key resulting in list element being skipped when `tabSelect: true`

AlexisDeschamps opened the pull request:
It seems the branch for handling a `tab` key event if `tabSelect: true` is missing an `evt.preventDefault()`, which results in an item being skipped when autocompletes are created in a list.

To fix this, I believe we could call `evt.preventDefault()` in all branches of the procedure, instead of only some.

I have manually tested that this fixes the problem on our UI. The existing tests also appear to pass. If this change seems to make sense, I can look into adding a test.

### Example
**Initial state**
![image](https://user-images.githubusercontent.com/12681350/121836647-bf2dcb00-cc88-11eb-8c6a-625e2657e453.png)

**(Current behavior): final state after pressing tab:**
![image](https://user-images.githubusercontent.com/12681350/121836881-395e4f80-cc89-11eb-979d-18431b6ccbeb.png)

**(Expected behavior): final state after pressing tab:**
![image](https://user-images.githubusercontent.com/12681350/121836792-074ced80-cc89-11eb-97e5-ed19c3fbd8f4.png)

AlexisDeschamps committed:
call preventDefault for all keydown branchese

LeaVerou reviewed (changes requested):
I would be wary about cancelling the event in every case, whether it's relevant or not. That could have serious accessibility consequences. For example, it disables tabbing even when `tabSelect` is not on, not to mention it cancels the event regardless of whether Tab was even pressed at all.
I'd welcome a PR that adds it under a more specific clause.

Thanks for contributing!

AlexisDeschamps committed:
add review updates

AlexisDeschamps commented:
@LeaVerou Thank you for the guidance. That makes sense. I modified the `preventDefault` call to be under a specific clause and added a unit test. Please let me know if there is something else I should add or change.

LeaVerou commented:
Merged, thanks!
